RecoEgamma/PhotonIdentification/plugins/PhotonIDValueMapProducer.cc

```diff
@@ -0,0 +1,527 @@
+								       ("esReducedRecHitCollectionMiniAOD"));
+
+  vtxToken_        = mayConsume<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("vertices"));
+  vtxTokenMiniAOD_ = mayConsume<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("verticesMiniAOD"));
+
+  // reco photons are castable into pat photons, so no need to handle reco/pat seprately
+  src_        = mayConsume<edm::View<reco::Photon> >(iConfig.getParameter<edm::InputTag>("src"));
+  srcMiniAOD_ = mayConsume<edm::View<reco::Photon> >(iConfig.getParameter<edm::InputTag>("srcMiniAOD"));
+
+  // The particleBasedIsolation object is relevant only for AOD, RECO format
+  particleBasedIsolationToken_ = mayConsume<edm::ValueMap<std::vector<reco::PFCandidateRef > > >
+    (iConfig.getParameter<edm::InputTag>("particleBasedIsolation"));
+  
+  // AOD has reco::PFCandidate vector, and miniAOD has pat::PackedCandidate vector.
+  // Both inherit from reco::Candidate, so we go to the base class. We can cast into
+  // the full type later if needed. Since the collection names are different, we
+  // introduce both collections
+  pfCandidatesToken_        = mayConsume< edm::View<reco::Candidate> >(iConfig.getParameter<edm::InputTag>("pfCandidates")); 
+  pfCandidatesTokenMiniAOD_ = mayConsume< edm::View<reco::Candidate> >(iConfig.getParameter<edm::InputTag>("pfCandidatesMiniAOD")); 
+
+  //
+  // Declare producibles
+  //
+  // Cluster shapes
+  produces<edm::ValueMap<float> >(phoFull5x5SigmaIEtaIEta_);  
+  produces<edm::ValueMap<float> >(phoFull5x5SigmaIEtaIPhi_);  
+  produces<edm::ValueMap<float> >(phoFull5x5E1x3_);  
+  produces<edm::ValueMap<float> >(phoFull5x5E2x2_);  
+  produces<edm::ValueMap<float> >(phoFull5x5E2x5Max_);  
+  produces<edm::ValueMap<float> >(phoFull5x5E5x5_);  
+  produces<edm::ValueMap<float> >(phoESEffSigmaRR_);  
+  // Isolations
+  produces<edm::ValueMap<float> >(phoChargedIsolation_);  
+  produces<edm::ValueMap<float> >(phoNeutralHadronIsolation_);  
+  produces<edm::ValueMap<float> >(phoPhotonIsolation_);  
+  produces<edm::ValueMap<float> >(phoWorstChargedIsolation_);  
+
+}
+
+PhotonIDValueMapProducer::~PhotonIDValueMapProducer() {
+}
+
+void PhotonIDValueMapProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+
+  using namespace edm;
+
+  // Constants 
+  const float coneSizeDR = 0.3;
+  const float dxyMax = 0.1;
+  const float dzMax  = 0.2;
+
+  edm::Handle<edm::View<reco::Photon> > src;
+
+  bool isAOD = true;
+  iEvent.getByToken(src_, src);
+  if( !src.isValid() ){
+    isAOD = false;
+    iEvent.getByToken(srcMiniAOD_,src);
+  }
+  if( !src.isValid() ) {
+    throw cms::Exception("IllDefinedDataTier")
+      << "DataFormat does not contain a photon source!";
+  }
+
+  // Configure Lazy Tools
+  if( isAOD )
+    lazyToolnoZS = new noZS::EcalClusterLazyTools(iEvent, iSetup, 
+						  ebReducedRecHitCollection_, 
+						  eeReducedRecHitCollection_,
+						  esReducedRecHitCollection_); 
```

Comment:
Please use a unique_ptr for lazyToolnoZS. There are "return" and "throw" statements before the "delete" that would leave lazyToolnoZS undeleted even after it is no longer needed.